controller/volume_controller.go

```diff
@@ -427,12 +427,31 @@ func (vc *VolumeController) ReconcileEngineReplicaState(v *longhorn.Volume, e *l
 		return nil
 	}
 
+	restoreStatusMap := map[string]*types.RestoreStatus{}
+	restoreCondition := types.GetCondition(v.Status.Conditions, types.VolumeConditionTypeRestore)
+	if restoreCondition.Status == types.ConditionStatusTrue && restoreCondition.Reason == types.VolumeConditionReasonRestoreInProgress {
+		replicaList := []*longhorn.Replica{}
+		for _, r := range rs {
+			replicaList = append(replicaList, r)
+		}
+		for addr, status := range e.Status.RestoreStatus {
+			rName := datastore.ReplicaAddressToReplicaName(addr, replicaList)
+			if _, exists := rs[rName]; exists {
+				restoreStatusMap[rName] = status
+			} else {
+				logrus.Warnf("Found unknown replica in the restore status map, address %v, restore status: %+v", addr, status)
+			}
+		}
+	}
+
 	// 1. remove ERR replicas
 	// 2. count RW replicas
 	healthyCount := 0
 	for rName, mode := range e.Status.ReplicaModeMap {
 		r := rs[rName]
-		if mode == types.ReplicaModeERR {
+		restoreStatus := restoreStatusMap[rName]
+		if mode == types.ReplicaModeERR ||
+			(restoreStatus != nil && restoreStatus.Error != "") {
```

Comment:
Do we have any log for the replicas marked as error due to restore error?

Reply:
There is already an error message reported by the engine monitor.